flowmachine/flowmachine/core/custom_query.py

```diff
@@ -33,13 +37,17 @@ class CustomQuery(Query):
     .table.Table for an equivalent that deals with simple table access
     """
 
-    def __init__(self, sql):
-        """
-
-        """
-
+    def __init__(self, sql: str, column_names: Union[List[str], Set[str]]):
         self.sql = pretty_sql(sql)
+        seen = {}  # Dedupe the column names but preserve order
+        self._column_names = [
+            seen.setdefault(x, x) for x in column_names if x not in seen
```

Comment:
Crikey, that's a lot of cheeky abuse of functionality in a single line... 😂

Reply:
No strong opinion, but I feel this sort of thing is often useful in a helper function - partly for reuse but mostly for readability. I like how marshmallow collects a few useful ones in their [utils module](), for example `is_iterable_but_not_string ()`. Naming this `dedupe_list_while_preserving_order()` would both eliminate the comment and allow the reader to not have to worry about the implementation. But again, no strong opinion if you want to keep this inline here.